scylla-macros/src/serialize/cql.rs

```diff
@@ -0,0 +1,281 @@
+
+        let rust_field_names = self
+            .ctx
+            .fields
+            .iter()
+            .map(|f| f.ident.as_ref().unwrap().to_string())
+            .collect::<Vec<_>>();
+        let udt_field_names = rust_field_names.clone(); // For now, it's the same
+        let field_types = self.ctx.fields.iter().map(|f| &f.ty).collect::<Vec<_>>();
+        let field_count = self.ctx.fields.len();
+
+        // Declare a helper lambda for creating errors
+        statements.push(self.ctx.generate_mk_typck_err());
+
+        // Check that the type we want to serialize to is a UDT
+        statements.push(
+            self.ctx
+                .generate_udt_type_match(parse_quote!(#crate_path::UdtTypeCheckErrorKind::NotUdt)),
+        );
+
+        // Generate a "visited" flag for each field
+        let visited_flag_names = rust_field_names
+            .iter()
+            .map(|s| syn::Ident::new(&format!("visited_flag_{}", s), Span::call_site()))
+            .collect::<Vec<_>>();
+        statements.extend::<Vec<_>>(parse_quote! {
+            #(let mut #visited_flag_names = false;)*
+        });
+
+        // Generate a variable that counts down visited fields.
+        statements.push(parse_quote! {
+            let mut remaining_count = #field_count;
+        });
+
+        // Generate a loop over the fields and a `match` block to match on
+        // the field name.
+        statements.push(parse_quote! {
+            for (field_name, field_type) in field_types {
+                match ::std::string::String::as_str(field_name) {
+                    #(
+                        #udt_field_names => {
+                            match <#field_types as #crate_path::SerializeCql>::preliminary_type_check(field_type) {
+                                ::std::result::Result::Ok(()) => {}
+                                ::std::result::Result::Err(err) => {
+                                    return ::std::result::Result::Err(mk_typck_err(
+                                        #crate_path::UdtTypeCheckErrorKind::FieldTypeCheckFailed {
+                                            field_name: <_ as ::std::clone::Clone>::clone(field_name),
+                                            err,
+                                        }
+                                    ));
+                                }
+                            };
+                            if !#visited_flag_names {
+                                #visited_flag_names = true;
+                                remaining_count -= 1;
+                            }
+                        }
+                    )*
+                    other => return ::std::result::Result::Err(mk_typck_err(
+                        #crate_path::UdtTypeCheckErrorKind::UnexpectedFieldInDestination {
+                            field_name: <_ as ::std::clone::Clone>::clone(field_name),
+                        }
+                    )),
```

Comment:
You don't seem to use `other` variable, isn't `_` enough? Or maybe you meant to use it?

Reply:
No, I'm not using it. I can change it to `_`.

I don't have a documentation link to support my thesis, but it looks like some warnings are not issued about the code generated by macros (e.g. about unused bindings).